### PR TITLE
Use service name in action confirmation popup

### LINKED ISSUE #8492: button confirmation question asking about the service, not the name
**Checklist**

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

**Describe the issue you are experiencing**

when the confirmation pops up, it displays some technical, not very clear question, 'are you sure you want to execute action  call-service?

**Describe the behavior you expected**

It would be a much better experience if the popup state: Do you want to activate the Alarm system, (given the fact the action name and name are provided

**Steps to reproduce the issue**

1.
2.
3.
...

**What version of Home Assistant Core has the issue?**

2021.2.3

**What was the last working version of Home Assistant Core?**

?

**In which browser are you experiencing the issue with?**

_No response_

**Which operating system are you using to run this browser?**

not relevant

**State of relevant entities**

```yaml
# Paste your state here.

```

**Problem-relevant frontend configuration**

```yaml
# Paste your YAML here.
      - type: button
        icon: mdi:shield-home
        name: ALARM SYSTEM
        tap_action:
          action: call-service
          service: script.turn_on_alarm_system
          confirmation: true
        action_name: Activate
```

**Javascript errors shown in your browser console/inspector**

```txt
# Paste your logs here.

```

Schermafbeelding 2021-02-28 om 23 06 51

Schermafbeelding 2021-02-28 om 23 07 00

am aware we can:
```
          confirmation:
            text: Are you sure you want to activate the Alarm system?
```

but this still feels like an unnecessary extra effort, where the frontend should be as human friendly as possible by default

### PR DESCRIPTION
**Pull request by spacegaier**
## Proposed change

If the action is `call-service` we now attempt to look-up the service name and show that in the confirmation popup.

Not all services in the backend provide a name currently (I created core PR https://github.com/home-assistant/core/pull/47204 to fix that for HA scripts and Python scripts), but that is not an issue. If there is no service name, we fallback to the current generic handling.

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes https://github.com/home-assistant/frontend/issues/8492
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [X] The code change is tested and works locally.
- [X] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/lovelace/common/handle-action.ts [hunk 1]
```diff
@@ -50,17 +50,28 @@ export const handleAction = async (
...
             "action",
-            hass.localize(
-              "ui.panel.lovelace.editor.action-editor.actions." +
-                actionConfig.action
-            ) || actionConfig.action
+            serviceName ||
+              hass.localize(
+                "ui.panel.lovelace.editor.action-editor.actions." +
+                  actionConfig.action
+              ) ||
```
Comment (bramkragten) [inline src/panels/lovelace/common/handle-action.ts:73]: We still want to show this part I think, just add what service will be called?

Comment (bramkragten) [inline src/panels/lovelace/common/handle-action.ts:73]:
```suggestion
              hass.localize(
                `ui.panel.lovelace.editor.action-editor.actions.${actionConfig.action}`
              ) ||
```

Comment (bramkragten) [reply src/panels/lovelace/common/handle-action.ts:73]: Also, I think many names assume that they will be used together with the domain/integration name, and not stand alone?

Comment (spacegaier) [reply src/panels/lovelace/common/handle-action.ts]: I think the end-user does not care if it is technically a "service" or not that is being called. So I opted only for the name.

I agree reg. the domain part. Need to see how to best include that.

Comment (spacegaier) [reply src/panels/lovelace/common/handle-action.ts:76]: @bramkragten I attempted to take over the logic that is also used in the ha-service-picker for the domain. The problem is that I cannot get the translations to get loaded. So the first time the popup gets shown, the domain is not translated, the second time around it is.

I attempted to manually force the "title" loading via `hass.loadBackendTranslation("title");` but I did not have any effect.

=> Any hints / tips?

Comment (bramkragten) [reply src/panels/lovelace/common/handle-action.ts:76]: Probably a timing issue then? Do you await it? And use its result as input for `domainToName`?

Comment (spacegaier) [reply src/panels/lovelace/common/handle-action.ts:77]: I am sure I did, but apparently both individually (either await or local var) and not combined (await + local var). Tried that now, and seems to work:

image

**Review by bramkragten — APPROVED** [review]

**Pull request merged**